turbinia/workers/file_system_timeline.py

```diff
@@ -68,43 +68,47 @@ def run(self, evidence, result):
     except dfvfs_errors.ScannerError as exception:
       status = 'Unable to open evidence: {0!s}'.format(exception)
       result.close(self, success=False, status=status)
+      return result
 
     # Iterate over all file entries and generate the output in bodyfile
     # format.
     try:
-      file_entries = None
-      with open(bodyfile_output, 'w', encoding='utf-8') as file_object:
-        file_entries = enumerate(entry_lister.ListFileEntries(base_path_specs))
-        while file_entries:
-          try:
-            _, (file_entry, path_segments) = next(file_entries)
-            bodyfile_entries = entry_lister.GetBodyfileEntries(
-                file_entry, path_segments)
-            for bodyfile_entry in bodyfile_entries:
-              file_object.write(bodyfile_entry)
-              file_object.write('\n')
-              number_of_entries += 1
-          except StopIteration:
-            break
-          except (dfvfs_errors.AccessError, dfvfs_errors.BackEndError,
-                  dfvfs_errors.MountPointError,
-                  dfvfs_errors.PathSpecError) as exception:
-            status = 'Unable to process file entry: {0!s}'.format(exception)
-            result.log(status)
-
-      if number_of_entries > 0:
-        output_evidence.number_of_entries = number_of_entries
-        result.add_evidence(output_evidence, evidence.config)
-        status = 'Generated file system timeline containing [{0:d}] entries'.format(
-            number_of_entries)
-        result.close(self, success=True, status=status)
-      else:
-        status = 'Unable to process any file entries.'
-        result.close(self, success=False, status=status)
-
+      bodyfile_fileobj = open(bodyfile_output, 'w', encoding='utf-8')
     except IOError as exception:
       status = 'Unable to create bodyfile local output file: {0!s}'.format(
           exception)
       result.close(self, success=False, status=status)
+      return result
+
+    if bodyfile_fileobj:
```

Comment:
If this didn't evaluate to `True` then that'd mean that the exception above has been raised and the result has been returned so probably not needed? In either case more of just a nitpick since shouldn't affect functionality.

Reply:
Done